### DISCUSSION TITLE
Calling endWith without arguments

### DISCUSSION THREAD

**Discussion by confraria** [ReactiveX/rxjs#6424] [General]
I was calling `endWith()` without any argument expecting it to be equivalent to `endWith(undefined)`.
As calling the operator has no effect, wouldn't it make sense to prevent the call with no arguments with the typescript signature?

**Comment by cartant** [comment]
Pretty much all of the rest parameter signatures in RxJS will accept zero arguments. This is by design, as it allows array spreading to be used:

```ts
const numbers: number[] = getSomeNumbers();
source.pipe(endWith(...numbers))
```

Requiring at least one argument would interfere with this and adds little value, IMO, as the behaviour that's effected by calling `endWith` or `concat` or `of` with zero arguments is consistent with how the code reads: `endWith()` ends with nothing.

**Reply by confraria** [reply]
🙈 oh of course. wasn't thinking about the rest parameter case.
thanks for pointing it out.
